runtime/lib/Runtime/OutgoingApiLimiter/index.ts

```diff
@@ -0,0 +1,40 @@
+import Runtime from '@/runtime/lib/Runtime';
+
+const MIN_NUMBER_OF_CALLS_TO_THROTTLE_PRIVATE = 5000;
+const MIN_NUMBER_OF_CALLS_TO_THROTTLE_PUBLIC = 1000;
+
+class OutgoingApiLimiter {
+  private REDIS_PREFIX = 'outgoing_api';
+
+  // in seconds for the uses count to reset
+  private EXPIRY_LENGTH = 60;
+
+  constructor(private runtime: Runtime) {}
+
+  private isPublic(): boolean {
+    return !this.runtime.authorization;
+  }
+
+  private makeRedisHostnameHash(hostname: string): string {
+    return `${this.REDIS_PREFIX}_${this.isPublic() ? 'public' : this.runtime.authorization!}_${hostname}`;
+  }
+
+  private async getHostnameUses(hostname: string): Promise<string | null> {
+    return this.runtime.services.redis.get(this.makeRedisHostnameHash(hostname));
+  }
```

Comment:
instead of writing `Number(getHostnameUses())` you should just alter this function to return `number | null` directly

```suggestion
  private async getHostnameUses(hostname: string): Promise<number | null> {
    const uses = await this.runtime.services.redis.get(this.makeRedisHostnameHash(hostname));
    
    return uses ? Number(uses) : null;
  }
```